astilectron.go

```diff
@@ -161,10 +166,8 @@ func (a *Astilectron) Start() (err error) {
 		return errors.Wrap(err, "provisioning failed")
 	}
 
-	// Unfortunately communicating with Electron through stdin/stdout doesn't work on Windows so all communications
-	// will be done through TCP
-	if err = a.listenTCP(); err != nil {
-		return errors.Wrap(err, "listening failed")
+	if err = a.establishConnection(); err != nil {
```

Comment:
I'd like to have this method return the type of listen it has started (`tcp` or `unix`) so that it's given to `a.execute()` below which in turns use it in the line that sets the `cmd`.

This line should become:

```go
var cmd = exec.CommandContext(ctx, a.paths.AppExecutable(), append([]string{a.paths.AstilectronApplication(), listenType, a.listener.Addr().String(), singleInstance}, a.options
.ElectronSwitches...)...)
```

That way, the type of listen is determined by GO only and it's sent to the JS through the flags.

Reply:
also `a.listener.Addr().String()` should become the path to the socket in case of listening through a unix socket

Reply:
working on this one